tests/boardfarm_plugins/boardfarm_prplmesh/tests/prplmesh_base_test.py

```diff
@@ -30,3 +32,143 @@ def prplmesh_status_check(self, entity_or_radio: Union[env.ALEntity, env.Radio])
+        self, msg: str, msg_type: int, eth_src: str, eth_dst: str = None, mid: int = None
+    ) -> [sniffer.Packet]:
+        """Verify that the wired sniffer has captured a CMDU.
+
+        Mark failure if the CMDU is not found.
+
+        Parameters
+        ----------
+        msg: str
+            Message to show in case of failure. It is formatted in a context like
+            "No CMDU <msg> found".
+
+        msg_type: int
+            CMDU message type that is expected.
+
+        eth_src: str
+            MAC address of the sender that is expected.
+
+        eth_dst: str
+            MAC address of the destination that is expected. If omitted, the IEEE1905.1 multicast
+            MAC address is used.
+
+        mid: int
+            Message Identifier that is expected. If omitted, the MID is not checked.
+
+        Returns
+        -------
+        [sniffer.Packet]
+            The matching packets.
+        """
+        debug("Checking for CMDU {} (0x{:04x}) from {}".format(msg, msg_type, eth_src))
+        result = self.dev.DUT.wired_sniffer.get_cmdu_capture_type(msg_type, eth_src, eth_dst, mid)
+        assert result, "No CMDU {} found".format(msg)
+        return result
+
+    def check_cmdu_type_single(
+        self, msg: str, msg_type: int, eth_src: str, eth_dst: str = None, mid: int = None
+    ) -> sniffer.Packet:
+        '''Like check_cmdu_type, but also check that only a single CMDU is found.'''
+        cmdus = self.check_cmdu_type(msg, msg_type, eth_src, eth_dst, mid)
+        assert len(cmdus) == 1, "Multiple CMDUs {} found".format(msg)
+        return cmdus[0]
+
+    def check_no_cmdu_type(
+        self, msg: str, msg_type: int, eth_src: str, eth_dst: str = None
+    ) -> NoReturn:
+        '''Like check_cmdu_type, but check that *no* machting CMDU is found.'''
+        debug("Checking for no CMDU {} (0x{:04x}) from {}".format(msg, msg_type, eth_src))
+        result = self.dev.DUT.wired_sniffer.get_cmdu_capture_type(msg_type, eth_src, eth_dst)
+        if result:
+            for packet in result:
+                debug("  {}".format(packet))
+            assert False, "Unexpected CMDU {}".format(msg)
+
+    def check_cmdu_has_tlvs(
+        self, packet: sniffer.Packet, tlv_type: int
+    ) -> [sniffer.Tlv]:
+        '''Check that the packet has at least one TLV of the given type.
+
+        Mark failure if no TLV of that type is found.
+
+        Parameters
+        ----------
+        packet: Union[sniffer.Packet, None]
+            The packet to verify. It may also be None, to make it easy to let it follow
+            check_cmdu_type_single. If it is None, no failure is raised. If it is not an IEEE1905
+            packet, a failure *is* raised.
+
+        tlv_type: int
+            The type of TLV to look for.
+
+        Returns
+        -------
+        [sniffer.Tlv]
+            List of TLVs of the requested type. Empty list if the check fails.
+        '''
+        assert packet, "No packet found"
```

Comment:
on one hand the description says:
`The packet to verify. It may also be None, to make it easy to let it follow check_cmdu_type_single. If it is None, no failure is raised.`
but on the other hand, you assert here.

Reply:
I'll fix the description in both test_flows.py and this file. I was careless and didn't read the descriptions properly when I changed it in test_flows :(